Source/Core/Core/HW/WiimoteEmu/Extension/TaTaCon.cpp

```diff
@@ -58,7 +60,20 @@ void TaTaCon::Update()
   // Flip button bits.
   tatacon_data.state ^= 0xff;
 
-  Common::BitCastPtr<DataFormat>(&m_reg.controller_data) = tatacon_data;
+  target_state->data = tatacon_data;
+}
+
+void TaTaCon::Update(const DesiredExtensionState& target_state)
```

Comment:
It looks like each extension reimplements this. I think the only difference is the `DataFormat` template argument? A helper function-template (taking the desired `DataFormat` as a template parameter) could be made to not have as much repetition.

Reply:
Mhm, that's a good idea. Lemme see if that's doable.